### PR TITLE
[thread-cert] refactor case 5.2.4 using pktverify

### PR DESCRIPTION
**Pull request by FiveDimensions**

### CODE REVIEW

File: tests/scripts/thread-cert/pktverify/packet_filter.py [hunk 1]
```diff
@@ -351,6 +351,21 @@ def _set_found_index(self, last_index: Tuple[int, int], index: Tuple[int, int]):
         if self._parent is not None:
             self._parent._set_found_index(last_index, index)
 
+    def filter_mle_advertisement(self, role, **kwargs):
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/pktverify/packet_filter.py:354]: nit: typing ?
```suggestion
    def filter_mle_advertisement(self, role: str, **kwargs):
```

File: tests/scripts/thread-cert/pktverify/packet_filter.py [hunk 2]
```diff
@@ -351,6 +351,21 @@ def _set_found_index(self, last_index: Tuple[int, int], index: Tuple[int, int]):
...
+
+        if role != 'REED':
+            tlv_set.add(consts.ROUTE64_TLV)
+
+        return self.filter_LLANMA().\
+            filter_mle_cmd(consts.MLE_ADVERTISEMENT).\
+            filter(lambda p: tlv_set ==
+                   set(p.mle.tlv.type) and\
+                   p.ipv6.hlim == 255
+                   )
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/pktverify/packet_filter.py:367]:
```suggestion
        return self.filter_LLANMA().\
            filter_mle_cmd(consts.MLE_ADVERTISEMENT).\
            filter(lambda p: tlv_set ==
                   set(p.mle.tlv.type) and\
                   p.ipv6.hlim == 255, **kwargs
                   )
```
pass **kwargs down?

File: tests/scripts/thread-cert/Cert_5_2_04_REEDUpgrade.py [hunk 3]
```diff
@@ -165,80 +211,228 @@ def test(self):
...
+        #         JITTER <= REED_ADVERTISEMENT_MAX_JITTER
+
+        pkts.filter_wpan_src64(REED).\
+            filter_mle_advertisement('REED').\
+            filter(lambda p:
+                   p.sniff_timestamp - _pkt.sniff_timestamp <=
+                   REED_ADVERTISEMENT_INTERVAL +
+                   REED_ADVERTISEMENT_MAX_JITTER and\
+                   p.sniff_timestamp - _pkt.sniff_timestamp >
+                   REED_ADVERTISEMENT_INTERVAL
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_5_2_04_REEDUpgrade.py:290]:
```suggestion
                   REED_ADVERTISEMENT_INTERVAL < p.sniff_timestamp - _pkt.sniff_timestamp <= REED_ADVERTISEMENT_INTERVAL + REED_ADVERTISEMENT_MAX_JITTER
```

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**